docs/guides/es2k/compiling-p4-programs.md

```diff
@@ -2,134 +2,107 @@
-
-# Compile p4 program
-p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
-            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
-            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
-            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
-            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bf-rt.json
-```
-
-The compiler will generate the following files:
-
-- simple_l3_l4_pna.p4info.txt
-- simple_l3_l4_pna.bf-rt.json
-- simple_l3_l4_pna.context.json
-- simple_l3_l4_pna.s
+# 
+1. Copy the entire p4-programs directory from the host to the P4Tools container
+#sudo docker cp p4-programs/ <p4Tools container id>:/opt/
+
+2. Set environment variables in the container.
+# Set the following env variables  in the container prior to building
+[root@a54d354e447e p4-programs]# which p4c
+/opt/p4-tools/p4c/bin/p4c
+[root@a54d354e447e p4-programs] export P4C_PATH=/opt/p4-tools/p4c/bin
+export CPT_PATH=/opt/p4-tools/cpt/bin
+export PATH=$P4C_PATH:$CPT_PATH:$PATH
+export LD_LIBRARY_PATH=$P4C_PATH/../lib:/usr/local/lib:$LD_LIBRARY_PATH
+export PREV_ARTIFACT_DIR=
+
+
+
+3. Compile l3-fwd_p2p,  all artifacts will be generated in artifacts directory.
+[root@a54d354e447e ~]# cd /opt/p4-programs
+[root@a54d354e447e p4-programs]# make l3-fwd_p2p
+# Compiling the /opt/p4-programs/layer-3-forwarding/l3-fwd_p2p/l3-fwd_p2p.p4 and generating the runtime files and assembly...
+/opt/p4-tools/p4c/bin/p4c --target idpf --arch pna -I/opt/p4-tools/p4c/bin/../lib -I/opt/p4-tools/p4c/bin/../share/p4c/p4include 
+-I/opt/p4-tools/p4c/bin/../share/p4c/idpf-lib --package-by-domain --p4runtime-files ./artifacts/l3-fwd_p2p/p4Info.txt --save-temps -Xp4c 
+"--Wdisable --no-pedantic --context ./artifacts/l3-fwd_p2p/context.json --bfrt ./artifacts/l3-fwd_p2p/bf-rt.json" 
+--save-temps --npic --format csr --pkg-version 1.2 --pkg-name "FXP Package" 
+-Xassembler ".cpt_ver.s" /opt/p4-programs/layer-3-forwarding/l3-fwd_p2p/l3-fwd_p2p.p4 
+-o artifacts/l3-fwd_p2p
+
+ 4. All the artifacts including the .pkg will be in the artifacts directory.
+[root@a54d354e447e p4-programs]# ls -lrt artifacts/l3-fwd_p2p/
+total 2884
+-rw-r--r--. 1 root root 950608 Dec 22 23:57 l3-fwd_p2p_0b32dac62c9b4c18b3213e04a6bb8c5b.pkgo
+-rw-r--r--. 1 root root   3045 Dec 23 00:18 p4Info.txt
+-rw-r--r--. 1 root root   8866 Dec 23 00:18 bf-rt.json
+-rw-r--r--. 1 root root  30390 Dec 23 00:18 context.json
+-rw-r--r--. 1 root root  37269 Dec 23 00:18 l3-fwd_p2p.s
+-rw-r--r--. 1 root root 950608 Dec 23 00:18 l3-fwd_p2p_b222a542c1474685bd70a36994d16101.pkgo
+-rw-r--r--. 1 root root 950608 Dec 23 00:18 l3-fwd_p2p.pkg 
 
+```
 These files are called _P4 artifacts_.
 
-## 5. Generating a Pkg File
-
-Use `cpt` to prepare the P4 artifacts for deployment:
-
-```bash
-cpt --npic --format csr --pbd  -o simple_l3_l4_pna.pkg \
-    cpt_ver.s simple_l3_l4_pna.s
-```
 
+## Deploying P4 programs
```

Comment:
```suggestion
## 4. Deploying the P4 program
```

1. If you're going to number the preceding headings, you need to number this one as well.
2. In the context of this document, explaining how to compile and deploy a _single_ P4 program.